pkg/controller/ec2/transitgateway/setup.go

```diff
@@ -0,0 +1,145 @@
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type custom struct {
+	kube   client.Client
+	client svcsdkapi.EC2API
+}
+
+func filterList(cr *svcapitypes.TransitGateway, obj *svcsdk.DescribeTransitGatewaysOutput) *svcsdk.DescribeTransitGatewaysOutput {
+	connectionIdentifier := aws.String(meta.GetExternalName(cr))
+	resp := &svcsdk.DescribeTransitGatewaysOutput{}
+	for _, TransitGateway := range obj.TransitGateways {
+		if aws.StringValue(TransitGateway.TransitGatewayId) == aws.StringValue(connectionIdentifier) {
+			resp.TransitGateways = append(resp.TransitGateways, TransitGateway)
+			break
+		}
+	}
+	return resp
+}
+
+func (e *custom) postObserve(_ context.Context, cr *svcapitypes.TransitGateway, obj *svcsdk.DescribeTransitGatewaysOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch awsclients.StringValue(obj.TransitGateways[0].State) {
+	case string(svcapitypes.TransitGatewayState_available):
+		cr.SetConditions(xpv1.Available())
+	case string(svcapitypes.TransitGatewayState_pending):
+		cr.SetConditions(xpv1.Creating())
+	case string(svcapitypes.TransitGatewayState_modifying):
+		cr.SetConditions(xpv1.Unavailable())
+	case string(svcapitypes.TransitGatewayState_deleting):
+		cr.SetConditions(xpv1.Deleting())
+	case string(svcapitypes.TransitGatewayState_deleted):
+		// TransitGateway is in status deleted - and really removed after 6 hours in aws
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	return obs, nil
+}
+
+func isUpToDate(cr *svcapitypes.TransitGateway, obj *svcsdk.DescribeTransitGatewaysOutput) (bool, error) {
+	return true, nil
+}
+
+func (e *custom) preCreate(ctx context.Context, cr *svcapitypes.TransitGateway, obj *svcsdk.CreateTransitGatewayInput) error {
+	// set external name as tag on the transit gateway
+	resType := "transit-gateway"
+	key := "crossplane-claim-name"
+	value := cr.ObjectMeta.Name
+
+	spec := svcsdk.TagSpecification{
+		ResourceType: &resType,
+		Tags: []*svcsdk.Tag{
+			{
+				Key:   &key,
+				Value: &value,
+			},
+		},
+	}
+	obj.TagSpecifications = append(obj.TagSpecifications, &spec)
+
+	return nil
+}
+
+func (e *custom) postCreate(ctx context.Context, cr *svcapitypes.TransitGateway, obj *svcsdk.CreateTransitGatewayOutput, cre managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	// set transit gateway id as external name annotation on k8s object after creation
+	meta.SetExternalName(cr, aws.StringValue(obj.TransitGateway.TransitGatewayId))
+	return cre, nil
+}
+
+// LateInitialize fills the empty fields in *svcapitypes.TransitGateway with
+// the values seen in svcsdk.DescribeTransitGatewaysOutput.
+// nolint:gocyclo
+func LateInitialize(cr *svcapitypes.TransitGatewayParameters, obj *svcsdk.DescribeTransitGatewaysOutput) error {
```

Comment:
We should make sure the CR does not have a value for given field before we late initialize. It seems we assign whatever we get from AWS, which could mean overriding user parameters.